app/src/split/bluetooth/service.c

```diff
@@ -24,12 +26,59 @@ LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 static uint8_t num_of_positions = ZMK_KEYMAP_LEN;
 static uint8_t position_state[POS_STATE_LEN];
 
+#define BEHAVIOR_LABEL_LEN 12
+
+static struct zmk_split_run_behavior_payload behavior_run_payload;
+
 static ssize_t split_svc_pos_state(struct bt_conn *conn, const struct bt_gatt_attr *attrs,
                                    void *buf, uint16_t len, uint16_t offset) {
     return bt_gatt_attr_read(conn, attrs, buf, len, offset, &position_state,
                              sizeof(position_state));
 }
 
+static ssize_t split_svc_run_behavior(struct bt_conn *conn, const struct bt_gatt_attr *attrs,
+                                      const void *buf, uint16_t len, uint16_t offset,
+                                      uint8_t flags) {
+    struct zmk_split_run_behavior_payload *payload = attrs->user_data;
+    uint16_t end_addr = offset + len;
+
+    LOG_DBG("offset %d len %d", offset, len);
+
+    if (end_addr > sizeof(struct zmk_split_run_behavior_payload)) {
+        return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
+    }
+
+    memcpy(payload + offset, buf, len);
+
+    // We run if:
+    // 1: We've gotten all the position/state/param data.
+    // 2: We have a null terminated string for the behavior device label.
+    if ((end_addr > sizeof(struct zmk_split_run_behavior_data)) &&
+        payload->behavior_dev[end_addr - sizeof(struct zmk_split_run_behavior_data)] == '\0') {
+        struct zmk_behavior_binding binding = {
+            .param1 = payload->data.param1,
+            .param2 = payload->data.param2,
+            .behavior_dev = payload->behavior_dev,
```

Comment:
It would be good to document somewhere as obvious as possible that if you're writing a behavior, you cannot count on binding->behavior_dev being valid after a handler returns, and if you need to save that data you should use `device_get_binding()` and save the device pointer instead (I think that's safe?) or `strdup` the string and free it when you're done with it.